eng/generators.targets

```diff
@@ -44,8 +44,12 @@
   <ItemGroup Condition="'@(EnabledGenerators)' != ''
                         and @(EnabledGenerators->AnyHaveMetadataValue('Identity', 'LibraryImportGenerator'))
                         and '$(IncludeLibraryImportGeneratorSources)' == 'true'">
-    <Compile Include="$(LibrariesProjectRoot)Common\src\System\Runtime\InteropServices\LibraryImportAttribute.cs" />
-    <Compile Include="$(LibrariesProjectRoot)Common\src\System\Runtime\InteropServices\StringMarshalling.cs" />
+
+    <!-- Only add the following files if we are not on the latest TFM. -->
+    <Compile Condition="'$(NetCoreAppCurrentTargetFrameworkMoniker)' != '$(TargetFrameworkMoniker)'"
+             Include="$(CoreLibSharedDir)System\Runtime\InteropServices\LibraryImportAttribute.cs" />
+    <Compile Condition="'$(NetCoreAppCurrentTargetFrameworkMoniker)' != '$(TargetFrameworkMoniker)'"
+             Include="$(CoreLibSharedDir)System\Runtime\InteropServices\StringMarshalling.cs" />
```

Comment:
I'm curious what's special about GeneratedMarshallingAttribute that below it's only included for .NET 7?

Reply:
That file has the attributes for the custom marshalling support (https://github.com/dotnet/runtime/issues/46838), which the generator doesn't support for down-level, so we didn't include it since it would basically be extra/unused. It should end up removed from this generators.targets file when we actually add those attributes to the runtime libraries.